Expand dict API to allow dtype strings

lucylq opened the pull request:
Summary:
Allow users to specify dtypes using strings, instead of kernel keys.

Note: this generates a 'fake' kernel key with no dim order data. I think this is OK, as I don't think ExecuTorch uses the kernel key for anything besides dtype.

https://www.internalfb.com/code/fbsource/[c02ea3c66de17a60b2f0e65462a03ccf5cba9804]/fbcode/caffe2/torchgen/selective_build/selector.py?lines=49-52

Differential Revision: D69335674

larryliu0820 reviewed (approved):
Thank you for adding the test

lucylq committed:
Expand dict API to allow dtype strings (#8323)

Summary:

Allow users to specify dtypes using strings, instead of kernel keys.

Note: this generates a 'fake' kernel key with no dim order data. I think this is OK, as I don't think ExecuTorch uses the kernel key for anything besides dtype.

https://www.internalfb.com/code/fbsource/[c02ea3c66de17a60b2f0e65462a03ccf5cba9804]/fbcode/caffe2/torchgen/selective_build/selector.py?lines=49-52

Reviewed By: larryliu0820

Differential Revision: D69335674
